### ISSUE TITLE
Bogus warning when using the address of anonymous data member

### ISSUE THREAD

**Issue by JohelEGP** [llvm/llvm-project#53296]
See https://godbolt.org/z/fMbYMMYKz. This was raised at https://gcc.gnu.org/bugzilla/show_bug.cgi?id=99571#c2.
```C++
struct I{
  union { int a{}; };
  int *i;
  constexpr I():i(&a){}
};
constexpr I i;
template<I>struct C{};
[[maybe_unused]]C<i>c;
```
```
<source>:4:20: warning: field '' is uninitialized when used here [-Wuninitialized]
  constexpr I():i(&a){}
                   ^
1 warning generated.
```
I'd expect something like `'<anynomous>.a'` instead of `''`, and `a` is certainly initialized.
